From bbc619d58d64f1eff44557721fba9ad6600802a2 Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Wed, 11 Jan 2023 15:06:45 +0530
Subject: [PATCH] ath12k: Remove paging_dyb.bin from paging.bin

The expected size for the paging.bin in the FW dump is only 6MB but
the parsed paging.bin is of 18MB. This is due to the addition of
paging_dyn.bin in the paging.bin in the coredump segment. Remove this
incorrect addition of paging_dyn.bin.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.h     |  3 ---
 drivers/net/wireless/ath/ath12k/coredump.c | 19 -------------------
 drivers/net/wireless/ath/ath12k/qmi.c      |  4 ----
 3 files changed, 26 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.h b/drivers/net/wireless/ath/ath12k/core.h
index 727be40..127c39c 100644
--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -1257,9 +1257,6 @@ void ath12k_core_halt(struct ath12k *ar);
 int ath12k_core_resume(struct ath12k_base *ab);
 int ath12k_core_suspend(struct ath12k_base *ab);
 int ath12k_config_qdss(struct ath12k_base *ab);
-int ath12k_coredump_mhi_update_bhie_table(struct ath12k_base *ab,
-					  void __iomem *va,
-					  phys_addr_t pa, size_t size);
 
 void ath12k_core_put_hw_group(struct ath12k_base *ab);
 
diff --git a/drivers/net/wireless/ath/ath12k/coredump.c b/drivers/net/wireless/ath/ath12k/coredump.c
index 2cf81d1..1b2ac30 100644
--- a/drivers/net/wireless/ath/ath12k/coredump.c
+++ b/drivers/net/wireless/ath/ath12k/coredump.c
@@ -330,25 +330,6 @@ out:
 	vfree(dump);
 }
 
-int ath12k_coredump_mhi_update_bhie_table(struct ath12k_base *ab,
-					  void __iomem *va,
-					  phys_addr_t pa, size_t size)
-{
-	struct ath12k_pci *ar_pci = (struct ath12k_pci *)ab->drv_priv;
-	struct mhi_controller *mhi_ctrl = ar_pci->mhi_ctrl;
-	int ret;
-
-	/* Attach Pageable region to MHI buffer so that it is
-	 * included as part of pageable region in dumps
-	 */
-	ret = mhi_update_bhie_table_for_dyn_paging(mhi_ctrl, va, pa, size);
-	if (ret)
-	        ath12k_dbg(ab, ATH12K_DBG_QMI,
-			   "failed to add Dynamic Paging region to MHI Buffer table %d\n", ret);
-
-	return ret;
-}
-
 void ath12k_coredump_m3_dump(struct ath12k_base *ab,
 			     struct ath12k_qmi_m3_dump_upload_req_data *event_data)
 {
diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index cf44426..edf80f4 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -3077,10 +3077,6 @@ static int ath12k_qmi_assign_target_mem_chunk(struct ath12k_base *ab)
 					ab->qmi.target_mem[i].size;
                         ab->qmi.target_mem[idx].type =
 					ab->qmi.target_mem[i].type;
-			ret = ath12k_coredump_mhi_update_bhie_table(ab,
-					ab->qmi.target_mem[idx].v.ioaddr,
-					ab->qmi.target_mem[idx].paddr,
-					ab->qmi.target_mem[idx].size);
 			sz += ab->qmi.target_mem[i].size;
 			if (ret < 0)
 				ath12k_warn(ab,
-- 
2.17.1

